crates/re_ui/src/lib.rs

```diff
@@ -445,6 +446,41 @@ impl ReUi {
         .inner
     }
 
+    /// Shows a `egui::DragValue` for a time in nanoseconds.
+    ///
+    /// The value and unit displayed adapts to the provided allowable time range.
+    #[allow(clippy::unused_self)]
+    pub fn time_drag_value(
+        &self,
+        ui: &mut egui::Ui,
+        value: &mut i64,
+        time_range: &RangeInclusive<i64>,
+    ) {
+        let span = time_range.end() - time_range.start();
+
+        let (unit, factor) = if span / 1_000_000_000 > 0 {
+            ("s", 1_000_000_000.)
+        } else if span / 1_000_000 > 0 {
+            ("ms", 1_000_000.)
+        } else if span / 1_000 > 0 {
+            ("μs", 1_000.)
+        } else {
+            ("ns", 1.)
+        };
```

Comment:
Tempted to reject this suggestion because:
- this would be a oddly specific `pub fn unit_from_duration_nano(duration: i64) -> (&'static str, f32)` function (not quite the "opposite" of `parse_duration`
- this would introduce a `re_format` dependency to `re_ui`, which I'd rather avoid

Reply:
The idea was for `re_format` to be a rather simple crate that we could depend on anywhere, but I see that dependencies on `arrow` and `comfy-table` has snuck in there 🤦  We should make those opt-in dependencies